### PR TITLE
move mbedtls/2.16.3 to dedicated folder

### PR DESCRIPTION
**Pull request by prince-chrismc**
Main reason is because on [the site](https://tls.mbed.org/download)
> In packaged form, mbed TLS 2.1.0 to mbed TLS 2.16 are available in both an Apache 2.0 licensed version (our primary open source license) and in a GPL 2.0 licensed version.
> Newer versions will be available under the Apache 2.0 license.
> You can find the most current stable version of mbed TLS on the Github releases page

Specify library name and version:  **mbedtls/2.16.3**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

My intention is to follow this up with a new version that's apache only, there's a lot less extra in the new recipe too

### CODE REVIEW

File: recipes/mbedtls/2.16.3/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.1)
```
Comment (prince-chrismc) [inline recipes/mbedtls/2.16.3/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.4)
```

**Comment by prince-chrismc** [comment]
Well this is a duplicate, which I did not find...

My problem is the diff, once [this suggestion](https://github.com/conan-io/conan-center-index/pull/2339/files#r471772715) is applied will muck up the recipe.

In terms on maintainability, if the different licensing schemes were separated I think it would be much easier to maintain. I was not thinking of how Conan handles the license attribute when I made this PR.

Since the other OP has vanished, I could continue what they started (add everything to the all folder) or I'll close that one in a separate PR when I added 2.23 in a new folder.

**Comment by prince-chrismc** [comment]
link https://github.com/conan-io/conan-center-index/pull/2339 so it's easy to find next time

**Comment by uilianries** [comment]
I don't think you need separate then only because the license, it's an simple attribute. Applying patches is more complicated.

**Comment by prince-chrismc** [comment]
follow up #2789

**Pull request closed without merging**